src/docgen/view/class.ts

```diff
@@ -10,6 +13,48 @@ import { StaticFunctions } from './static-functions';
 const CONSTRUCT_CLASS = 'constructs.Construct';
 
 export class Class {
+  public static toMarkdown(
```

Comment:
It looks like this class now holds both 

* Conversion to JSON (member function)
* MarkDown rendering (static function)

This seems like 2 distinct concerns, and feels like they shouldn't be combined. The fact that the MarkDown conversion methods are stateless also requires passing around the `MarkdownRenderContext` object everywhere.

Why don't we make the Markdown rendering a (visitor) class instead:

```ts
public class MarkdownRenderer {
  private readonly document = new MarkdownDocument();

  constructor(private readonly linkFormatter, private readonly otherFormatter, private readonly someOtherContext) {
  }

  public visitClass(klass: ClassSchema) {
    this.document.h2(klass.name);
    this.document.paragraph(klass.docs.summary);
    this.document.paragraph(klass.docs.remarks);

    for (const method of klass.methods) {
      this.visitMethod(method);
    }
  }

  public visitMethod(method: MethodSchema) { }
}
```

This keeps all the MD concerns nicely colocated, and all renderers can just happily share state and private helpers.